### ISSUE TITLE
suggestion: stylua formatter for Lua

### ISSUE THREAD

**Issue by docwhat** [trunk-io/plugins#214]
I added a [`stylua`](https://github.com/JohnnyMorganz/StyLua#readme) linter to [my `.trunk.yaml`](https://github.com/docwhat/dotfiles/blob/main/.trunk/trunk.yaml) and thought it worth inclusion.

I don't have the time to massage it into shape for a PR but it mostly works.

Here are the relavant parts:

```yaml
version: 0.1
cli:
  version: 1.6.1
lint:
  files:
    - name: lua
      extensions:
        - lua
      shebangs:
        - lua
  downloads:
    - name: stylua
      version: 0.17.0
      downloads:
        - os:
            linux: linux
            macos: macos
          cpu:
            x86_64: x86_64
            arm_64: aarch64
          url: https://github.com/JohnnyMorganz/StyLua/releases/download/v${version}/stylua-${os}-${cpu}.zip
  enabled:
    - stylua@0.17.0
  definitions:
    - name: stylua
      files: [lua]
      download: stylua
      commands:
        - formatter: true
          output: rewrite
          run: stylua --verify --search-parent-directories --stdin-filepath ${target} -
          stdin: true
          success_codes: [0, 1]
      run_linter_from: workspace
      good_without_config: true
      direct_configs:
        - .stylua.toml
      environment:
        - name: PATH
          list: ["${linter}"]
      version_command:
        parse_regex: ${semver}
        run: stylua --version
```

**Comment by TylerJang27** [comment]
Thanks for bringing this up, as well as for your continued contributions! We'll look into getting this integrated soon; it'll be good to add some support for Lua too.

**Comment by TylerJang27** [comment]
@docwhat I just opened up #222, feel free to take a look and if anything seems off lmk. I plan to merge it sometime later tomorrow, fyi. Thanks for the suggestion; it was already 99% complete!

**Comment by TylerJang27** [comment]
Landed in #222. Will be included in the next release!
